lib/smb2/packet.rb

```diff
@@ -47,17 +47,134 @@ class InvalidFlagError < StandardError; end
-  QUERY_INFO_TYPES = {
-    # SMB2_0_INFO_FILE
-    FILE: 0x01,
-    # SMB2_0_INFO_FILESYSTEM
-    FILESYSTEM: 0x02,
-    # SMB2_0_INFO_SECURITY
-    SECURITY: 0x03,
-    # SMB2_0_INFO_QUOTA
-    QUOTA: 0x04
+  # Used in {CreateRequest#disposition}
+  CREATE_DISPOSITIONS = {
+    # If the file already exists, supersede it. Otherwise, create the file.
+    # This value SHOULD NOT be used for a printer object.<30>
+    FILE_SUPERSEDE: 0x00000000,
+    # If the file already exists, return success; otherwise, fail the
+    # operation. MUST NOT be used for a printer object.
+    FILE_OPEN: 0x00000001,
+    # If the file already exists, fail the operation; otherwise, create the
+    # file.
+    FILE_CREATE: 0x00000002,
+    # Open the file if it already exists; otherwise, create the file. This
+    # value SHOULD NOT be used for a printer object.<31>
+    FILE_OPEN_IF: 0x00000003,
+    # Overwrite the file if it already exists; otherwise, fail the operation. MUST
+    # NOT be used for a printer object.
+    FILE_OVERWRITE: 0x00000004,
+    # Overwrite the file if it already exists; otherwise, create the file.
+    # This value SHOULD NOT be used for a printer object.<32>
+    FILE_OVERWRITE_IF: 0x00000005,
+  }.freeze
+
+  # Used in {CreateRequest#create_options}
+  #
+  # https://msdn.microsoft.com/en-us/library/cc246502.aspx
+  CREATE_OPTIONS = {
+    FILE_DIRECTORY_FILE: 0x0000_0001,
+    FILE_WRITE_THROUGH: 0x0000_0002,
+    FILE_SEQUENTIAL_ONLY: 0x0000_0004,
+    FILE_NO_INTERMEDIATE_BUFFERING: 0x0000_0008,
+    FILE_SYNCHRONOUS_IO_ALERT: 0x0000_0010,
+    FILE_SYNCHRONOUS_IO_NONALERT: 0x0000_0020,
+    FILE_NON_DIRECTORY_FILE: 0x0000_0040,
+    FILE_COMPLETE_IF_OPLOCKED: 0x0000_0100,
+    FILE_NO_EA_KNOWLEDGE: 0x0000_0200,
+    FILE_RANDOM_ACCESS: 0x0000_0800,
+    FILE_DELETE_ON_CLOSE: 0x0000_1000,
+    FILE_OPEN_BY_FILE_ID: 0x0000_2000,
+    FILE_OPEN_FOR_BACKUP_INTENT: 0x0000_4000,
+    FILE_NO_COMPRESSION: 0x0000_8000,
+    FILE_OPEN_REMOTE_INSTANCE: 0x0000_0400,
+    FILE_OPEN_REQUIRING_OPLOCK: 0x0001_0000,
+    FILE_DISALLOW_EXCLUSIVE: 0x0002_0000,
+    FILE_RESERVE_OPFILTER: 0x0010_0000,
+    FILE_OPEN_REPARSE_POINT: 0x0020_0000,
+    FILE_OPEN_NO_RECALL: 0x0040_0000,
+    FILE_OPEN_FOR_FREE_SPACE_QUERY: 0x0080_0000,
+  }.freeze
+
+  # Used in {CreateRequest#desired_access} when opening a file, pipe, printer.
+  # For access mask values to use when opening a directory, see
+  # {DIRECTORY_ACCESS_MASK}.
+  #
+  # From the documentation ([2.2.13.1.1 File_Pipe_Printer_Access_Mask](https://msdn.microsoft.com/en-us/library/cc246802.aspx)):
+  #
+  # > The SMB2 Access Mask Encoding in SMB2 is a 4-byte bit field value that
+  #   contains an array of flags. An access mask can specify access for one of
+  #   two basic groups: either for a file, pipe, or printer (specified in
+  #   section 2.2.13.1.1) or for a directory (specified in section 2.2.13.1.2).
+  #
+  # @see DIRECTORY_ACCESS_MASK
+  FILE_ACCESS_MASK = {
```

Comment:
Similar to https://github.com/rapid7/smb2/pull/30/files#r29786993, would showing the shifts make the derivation clearer?